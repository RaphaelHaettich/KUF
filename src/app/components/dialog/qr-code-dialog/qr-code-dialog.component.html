<div mat-dialog-content class="{{ componentName }}__header">
    <h2 class="{{ componentName }}__title" mat-dialog-title>{{ data.title }}</h2>
</div>
<div mat-dialog-content class="{{ componentName }}__content">
    <span>{{ data.subTitle }}</span>
    <qrcode [qrdata]="data.qrString" [width]="256" [errorCorrectionLevel]="'M'"></qrcode>
</div>
<div mat-dialog-actions class="{{ componentName }}__actions">
    <button (click)="closeDialog()" type="button" mat-button obButton="primary" mat-dialog-close>
        {{ "i18n.common.close" | translate }}
    </button>
</div>
