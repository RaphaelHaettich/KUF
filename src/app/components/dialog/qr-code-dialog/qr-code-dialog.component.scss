@import '@oblique/oblique/styles/scss/core/mixins/typography';
@import '@oblique/oblique/styles/scss/core/mixins/layout';
@import '@oblique/oblique/styles/scss/core/variables';

.qr-code-dialog {
    &__title {
        font-size: $ob-font-size-xl !important;
    }

    &__header {
        display: flex;
        justify-content: center;
    }

    &__content {
        display: flex;
        align-items: center;
        flex-direction: column;
    }
    
    &__actions {
        display: flex;
        justify-content: center;
    }

}
