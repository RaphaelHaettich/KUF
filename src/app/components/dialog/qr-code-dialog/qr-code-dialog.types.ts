export type QrShowData = {
    title: string;
    qrString: string;
    subTitle: string;
};
