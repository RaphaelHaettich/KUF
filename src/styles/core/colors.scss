$kuf-red: #dc0018;
$kuf-error: #C70023;
$kuf-yellow: #FF9B00;
$kuf-green: #00813a;
$kuf-grey: #808080;
$kuf-lightgrey: #E5E5E5;
$kuf-blue: #006699;
