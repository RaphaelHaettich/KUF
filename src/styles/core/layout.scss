#content {
    padding: 0.3rem !important;
    background-color: rgba(216, 216, 216, 0.603);
}
