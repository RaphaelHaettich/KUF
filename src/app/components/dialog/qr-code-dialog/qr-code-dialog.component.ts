import {Component, inject} from '@angular/core';
import {MAT_DIALOG_DATA, MatDialog} from '@angular/material/dialog';
import {QrShowData} from './qr-code-dialog.types';

@Component({
    selector: 'app-qr-code-dialog',
    templateUrl: './qr-code-dialog.component.html',
    styleUrl: './qr-code-dialog.component.scss',
})
export class QrCodeDialogComponent {
    componentName = 'qr-code-dialog';
    readonly data = inject<QrShowData>(MAT_DIALOG_DATA);
    private readonly dialog = inject(MatDialog);

    public closeDialog() {
        this.dialog.closeAll();
    }
}
