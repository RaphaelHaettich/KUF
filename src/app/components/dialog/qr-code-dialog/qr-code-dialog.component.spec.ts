import {ComponentFixture, TestBed} from '@angular/core/testing';
import {MAT_DIALOG_DATA, MatDialog, MatDialogModule} from '@angular/material/dialog';
import {QrCodeDialogComponent} from './qr-code-dialog.component';
import {QRCodeModule} from 'angularx-qrcode';
import {TranslateModule} from '@ngx-translate/core';
import {multiTranslateLoader} from '@oblique/oblique';
import {HttpClientModule} from '@angular/common/http';

describe('QrCodeDialogComponent', () => {
    let component: QrCodeDialogComponent;
    let fixture: ComponentFixture<QrCodeDialogComponent>;
    let dialog: MatDialog;

    beforeEach(async () => {
        await TestBed.configureTestingModule({
            declarations: [QrCodeDialogComponent],
            imports: [
                MatDialogModule,
                QRCodeModule,
                TranslateModule.forRoot(multiTranslateLoader()),
                HttpClientModule,
            ],
            providers: [
                {
                    provide: MAT_DIALOG_DATA,
                    useValue: {
                        qrString: 'qrString',
                        title: 'title',
                        subTitle: 'subTitle',
                    },
                },
            ],
        }).compileComponents();
    });

    beforeEach(() => {
        fixture = TestBed.createComponent(QrCodeDialogComponent);
        component = fixture.componentInstance;
        dialog = TestBed.inject(MatDialog);
        fixture.detectChanges();
    });

    it('should create the component', () => {
        expect(fixture).toMatchSnapshot();
    });

    it('should close the dialog', () => {
        jest.spyOn(dialog, 'closeAll');
        component.closeDialog();
        expect(dialog.closeAll).toHaveBeenCalled();
    });
});
