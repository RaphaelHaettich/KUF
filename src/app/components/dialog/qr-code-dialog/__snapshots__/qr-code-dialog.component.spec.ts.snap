// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QrCodeDialogComponent should create the component 1`] = `
<app-qr-code-dialog
  componentName={[Function String]}
  data={[Function Object]}
  dialog={[Function _MatDialog]}
>
  <div
    class="mat-mdc-dialog-content mdc-dialog__content qr-code-dialog__header"
    mat-dialog-content=""
  >
    <h2
      class="mat-mdc-dialog-title mdc-dialog__title qr-code-dialog__title"
      mat-dialog-title=""
    >
      title
    </h2>
  </div><div
    class="mat-mdc-dialog-content mdc-dialog__content qr-code-dialog__content"
    mat-dialog-content=""
  >
    <span>
      subTitle
    </span>
    <qrcode>
      <div
        class="qrcode"
      />
    </qrcode>
  </div><div
    class="mat-mdc-dialog-actions mdc-dialog__actions qr-code-dialog__actions"
    mat-dialog-actions=""
  >
    <button
      mat-button=""
      mat-dialog-close=""
      obbutton="primary"
      type="button"
    >
       i18n.common.close 
    </button>
  </div>
</app-qr-code-dialog>
`;
